### DISCUSSION TITLE
使用简悦 + Obsidian 打造一站式知识管理方案

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#1631] [服务]
![rHjKd1.png](https://s3.ax1x.com/2020/12/29/rHjKd1.png)

> 知识管理有很多种方式，这里提一种：**如何通过简悦 + Obsidian 进行拆书**。

### 优势

> 简悦 2.1 与 Obsidian 都可以基于本地系统，所以 **你的数据你掌握，且放在你需要的地方**，进行全平台同步。

### 什么是拆书

> 拆书就是转述者或讲书人能把书中的内容、精华提炼出来，然后将原著的中心思想和精华表达具体清楚；而对于我来说，拆书相当于一种将书看薄看熟练的过程。 via [利用 Obsidian 拆书](https://zhuanlan.zhihu.com/p/338719525)

### 准备工作

1. 简悦需要 **[高级账户](https://simpread.pro/price.html)**
2. 配置并运行 **[简悦 · 同步助手](http://ksria.com/simpread/docs/#/Sync)**
3. 需将 **[简悦 · 同步助手导出服务](http://ksria.com/simpread/docs/#/Sync?id=导出服务)** 的文件夹设置为 Obsidian 库文件夹（或其子文件夹）

### 思路如下

1. 在网页使用简悦的标注进行快速标注（简悦支持连续标注），而这个过程类似拆书。
2. 之后将这些快速标注的内容通过 [简悦 · 同步助手](http://ksria.com/simpread/docs/#/Sync?id=导出服务) 导出到 Obsidian 的目录（以 Markdown 的形式）
3. 同时在标注时，也可以做一些备注，标签的操作（如果只是快速拆书的话，这些可后期再做），甚至于可以将某条标注单独导出。
4. 快速标注后，会在简悦的稍后读生成一个基于标注的页面，在这个页面可以集中管理这些标注，以便做二次修改（补充），同时这些内容也可以导出为 Markdown。
5. 最后每天通过这种方式产生的稍后读（标注）会已邮件汇总的方式发给你指定的邮箱（需配置），用于当天的 [加深回顾](http://ksria.com/simpread/docs/#/每日回顾)。
6. 这篇稍后读（包含若干通过拆书而产生的标注）也可以通过 发送邮件 / 发送到 Kindle / 导出的 PDF 的形式 / 导出 Epub 的形式来辅助你对这篇文章的加深记忆。

### 相比其它标注功能

> 因为上面提到了标注，所以跟其它标注功能相比。

1. 简悦支持单条标注导出（包括：导出到本地和其它生产力工具），所以省去了 **复制/粘贴** 的操作。

2. 简悦的数据文件都在本地（稍后读和配置文件），所以与 Obsidian 天然搭配。

### 利用简悦补齐 KPM 前几个环节

因为很喜欢 Obsidian 所以多说点 **方法论** 的事情。  😀 

> 一般来说使用 Obsidian 都是为了补齐 PKM 中的最后一环： **内化**

而 KPM 的前几个环节：**信息的收集** 和 **信息的过滤**（有效信息）我一般都使用了下面的方案：

1. 使用 Inoreader 过滤功能来筛除一批无效的噪音。via [XDash Weblog](http://xdash.one/use-inoreader-to-filter-news.html)

2. 留下来的自然是需要的内容（信息）。

  > 通过简悦 2.1 提供的 [RSS 阅读器辅助方式](http://ksria.com/simpread/docs/#/RSSReader) 或 [此方式](https://github.com/Kenshin/simpread/discussions/1559) 将这类信息直接导入到简悦的稍后读。

3. 这些信息（还不能称之为知识），大致包括如下几类：

    - 只是个人感兴趣的信息，但不具有反复理解的内容。

      > 我会把它们通过简悦 [发送到邮件](http://ksria.com/simpread/docs/#/Sync?id=%e9%82%ae%e4%bb%b6%e6%9c%8d%e5%8a%a1)，利用碎片时间在手机上阅读（阅读后会删除它，因为只是用来阅读。（当然，觉得有意义的话，可以使用离线 HTML 的方式，将它永久的保存下来。

   - 有意义的信息

      > 通过上面描述的拆书方式，发送到 Obsidian 做以后 **内化** 用（内化的结果是：将这些信息转换为你的知识

    - 零碎但具有收集意义的信息

      > 比如：我要去旅行，在这个前提下，会收集旅行相关的内容，这些内容并不具有知识的属性，但确实是你需要的内容，且它们都分散在不同的页面。

      > 遇到这种情况的话，我会通过简悦 [合辑的功能](http://ksria.com/simpread/docs/#/%E7%A8%8D%E5%90%8E%E8%AF%BB?id=%e5%90%88%e8%be%91) 将有需要的内容整理到一个 **页面**，[这是一个合辑的例子](https://simpread.pro/s/z2Axt6i1)

      > 同样即便是合辑，也支持稍后读的全部操作，如：导出 Markdown 等等

### 使用简悦如何进行卢曼的卡片笔记法 （Zettelkasten）：

> Zettelkasten 将笔记分为以下几个方面：

1. Fleeting notes → 阅读资料，并在过程中高亮或者会随手记下临时笔记
2. Literature notes → 将高亮的部分或者临时笔记总结为文献笔记
3. Permanent notes → 用自己的话将文献笔记转述为永久笔记

> 实现它们的思路：

1. Fleeting notes → 简悦的标注
2. Literature notes → 简悦的合辑（或对某个稍后读的多个标注进行详细备注的操作，简悦的标注与稍后读具有一样的地位和结构）
3. Permanent notes → 使用 Obsidian

### 最后

在这里只是抛砖引玉的说了下 简悦 与 Obsidian 的配合方案，这也仅仅只是其中一种而已。因为简悦 与 Obsidian 都天然支持本地文件系统，所以基于此可以实现更多的联动，欢迎有更多关于此的使用方案。

### 引申

> **[简悦 2.2.0](http://ksria.com/simpread/welcome/version_2.2.0.html)** 发布后，对双链笔记的支持增强了很多，你可以在 https://github.com/Kenshin/simpread/discussions/categories/服务 中查看全部跟双链笔记相关的教程。

🙏

**Comment by Sailfishc** [comment]
赞，思路很棒

**Reply by Kenshin** [reply]
谢谢，这只是抛砖引玉而已。😄

**Reply by 973339348** [reply]
添加单挑笔记 略显麻烦 请问下有没有一些解决的办法吗？

**Reply by Kenshin** [reply]
> 添加单挑笔记 略显麻烦 请问下有没有一些解决的办法吗？

![image](https://user-images.githubusercontent.com/81074/130345106-00d58475-a8c0-4e83-a0a4-e40a3f537582.png)

应该没有比这个方式更简单的了。

**Comment by zzxbisheng** [comment]
感谢大大们，个人有个小想法，就是在通过简阅导出笔记和标注的时候，可以附带标注在文章为知链接，这样的话导出到笔记软件的时候可以更好的溯源，也方便修改

**Reply by Kenshin** [reply]
嗨，可以考虑，谢谢建议。

**Reply by Kenshin** [reply]
嗨，补充下，快速回源的话，看看这篇文章 https://zhuanlan.zhihu.com/p/345900114

**Reply by zzxbisheng** [reply]
这个方法确实不错，但增加了繁琐程度，因为增加了工具和步骤。
再就是我发现标注的默认排序是按照添加标注的时间的，然而实际使用中可能更多人是需要按照前后文顺序进行默认排序的，希望能够增加相关设置！

**Reply by Kenshin** [reply]
关于操作步骤的问题会在新版本中解决掉，这么做的好处是：可以将页面真正的离线保存。

标注可以通过拖拽方式进行手动排序。（后续会考虑自动顺序的方案）

**Comment by Joilence** [comment]
也有这个想法，但是简阅有定义导出 Markdown 格式的功能么？现在格式是这样：

```
> 本文由 [简悦 SimpRead](http://ksria.com/simpread/) 转码， 原文地址 [title](url.com)

> highlight
> 
> 备注：  
> note
```

希望可以把格式转换成这样的：

```
[Title](url.com)

> highlight

note

> highlight

note
```

**Reply by Kenshin** [reply]
嗨，有解决方案，但没更新文档。
更新后我会通知你。

**Reply by Joilence** [reply]
@Kenshin 好的，谢谢！

**Reply by Kenshin** [reply]
嗨，抱歉，今天一直在处理文档的事情，耽搁了。

- 简悦 2.2.0 文档已全部更新，细节请看 https://github.com/Kenshin/simpread/issues/2158#issuecomment-845868642

- 简悦 2.2.0 发布后产生了一些通用性问题，细节请看 https://github.com/Kenshin/simpread/issues/2158

**Reply by Kenshin** [reply]
关于你提的问题，后续我会考虑出个教程，但现在只能看下文档 [定制化导出](http://ksria.com/simpread/docs/#/%E5%AE%9A%E5%88%B6%E5%8C%96%E5%AF%BC%E5%87%BA?id=markdown)

你看看能否可以先试试，等后续教程出了后我再通知你。

**Reply by Kenshin** [reply]
@Joilence 

嗨，新文档已经出来，因为太多，所以不一一介绍了，你可以在 https://github.com/Kenshin/simpread/discussions/categories/服务 里面查看全部跟双链笔记相关的内容。

**Comment by ernestyao** [comment]
能否导出到obsidian时，将图片单独下载到一个子文件夹？嵌入markdown会导致文件略大，用其他工具打开md文件加载可能会慢，感觉也不够“优雅”。obsidian里对文件夹支持还不错，我一般会设一个附件文件夹，只要将随md文件导出的图片文件夹挪过去，一般链接就不会失效。

**Reply by Kenshin** [reply]
嗨，可以试试 https://github.com/Kenshin/simpread/discussions/2241 
如果是 Windows 系统的话

**Reply by ernestyao** [reply]
> 嗨，可以试试 #2241
> 如果是 Windows 系统的话

MacOS......

**Reply by Kenshin** [reply]
你可以这样，用同步助手导出 textbundle 然后手动删除后缀，就是个文件夹，打开它，有个 .markdown 结尾的文件（改名为 .md ），这个结构就能用在 ob 了。

1.0.2 版同步助手会增加你说的功能。

**Reply by ernestyao** [reply]
> 你可以这样，用同步助手导出 textbundle 然后手动删除后缀，就是个文件夹，打开它，有个 .markdown 结尾的文件（改名为 .md ），这个结构就能用在 ob 了。
> 
> 1.0.2 版同步助手会增加你说的功能。

多谢指点🙏

**Reply by Kenshin** [reply]
嗨，我把你的这个诉求和解决方案，记录到了 https://github.com/Kenshin/simpread/discussions/2360

**Reply by Aspyrine** [reply]
> 你可以这样，用同步助手导出 textbundle 然后手动删除后缀，就是个文件夹，打开它，有个 .markdown 结尾的文件（改名为 .md ），这个结构就能用在 ob 了。
> 
> 1.0.2 版同步助手会增加你说的功能。

但是这样的话md文档里面的图片引用就要修改过了，另外md文档的标题和asse文件夹也要重命名，所以想问下有没有更好的和ob互动的方式呢？

**Reply by Kenshin** [reply]
@Aspyrine 

有的，如果是 Obsidian 和简悦深度使用者，可以试试这个专门为 Obsidian 开发的插件 https://github.com/Kenshin/simpread/discussions/2889 （目前为内测状态，所以没有对外宣传）

**Reply by ernestyao** [reply]
这个插件说明没看出图片是怎么处理的，其实保存过程中最麻烦的是下载图片并本地化。textbundle的方式虽然手工，但图片保留较好。

**Reply by Kenshin** [reply]
> 这个插件说明没看出图片是怎么处理的，其实保存过程中最麻烦的是下载图片并本地化。textbundle的方式虽然手工，但图片保留较好。

这个插件并不是用于保存数据到 obsidian 的，所以跟你说的无关。

保存图片的话，还是 textbundle 方便。

**Reply by dwSun** [reply]
我自己写了一个脚本，把内嵌的图片抽取出来放到 ob 的 asset 文件夹里面。

**Reply by Kenshin** [reply]
@dwSun 

不错啊，什么形式的脚本？能否贡献出来？

**Reply by dwSun** [reply]
[extract_imgs.py.txt](https://github.com/Kenshin/simpread/files/7853517/extract_imgs.py.txt)

**Reply by Kenshin** [reply]
@dwSun 

哈哈，不错啊。你也可以试试使用 uTools 的联动方式 e.g. https://github.com/Kenshin/simpread/discussions/3237 看看对你是否有帮助？（操作起来更简单些）

**Reply by dwSun** [reply]
ob有个插件 Local Images 也可以实现类似的功能，不过我的脚本是用了文件名加图片hash。我个人比较习惯由ob管理双链，而不是使用目录。

**Reply by ernestyao** [reply]
> ob有个插件 Local Images 也可以实现类似的功能，不过我的脚本是用了文件名加图片hash。我个人比较习惯由ob管理双链，而不是使用目录。

Local Image好像没法这么用，这个插件是粘贴链接的时候下载，直接打开一个.md文件不能出发下载。

**Reply by Kenshin** [reply]
@dwSun 

我说的 uTools 这个插件可以通过 `选中 .md` 下载。

**Reply by angel126** [reply]
> ob有个插件 Local Images 也可以实现类似的功能，不过我的脚本是用了文件名加图片hash。我个人比较习惯由ob管理双链，而不是使用目录。

可否分享一下脚本？谢谢。

**Reply by dwSun** [reply]
> 

上面有

**Comment by YangWaldon** [comment]
希望能增加导出为org格式的笔记文档的功能，以支持logseq的org-mode

**Reply by Kenshin** [reply]
嗨，谢谢建议。

不太了解这种结构，不知道有什么方面的文档给我参考下，尤其是 md → org-mode 的方案。

**Reply by Kenshin** [reply]
之前忘记通知了，简悦已经支持 `org-mode` 的导出，细节 [看这里](https://github.com/Kenshin/simpread/discussions/3802)。

**Comment by YangWaldon** [comment]
了解到您也在logseq群里，logseq的org mode的org文件的格式好像大部分只是block的开头的定义格式还有图片的引用格式与md不同，org mode应该也是参考emacs-org mode 做的吧

**Reply by Kenshin** [reply]
嗨，因为你的通过邮箱回复的，所以格式结构乱掉了，要是原因的话，请前往 https://github.com/Kenshin/simpread/discussions/1631#discussioncomment-1236582 登录并回复。

**Comment by quanru** [comment]
标注后，得手动点击导出，才能到本地吗？

**Reply by Kenshin** [reply]
使用自动化功能

**Reply by ryzzn** [reply]
> 

不知道能不能有快捷键触发自动化的方案？比如我想使用简阅收集文章，并且同步到本地。现在的方式我必须按AA然后按DD才能触发，有方法可以省略成一步么？

**Reply by Kenshin** [reply]
> 不知道能不能有快捷键触发自动化的方案？比如我想使用简阅收集文章，并且同步到本地。现在的方式我必须按AA然后按DD才能触发，有方法可以省略成一步么？

没有的，简悦是需要进入阅读模式后再使用的，我希望有深度阅读的页面才可以有后面的事情，如稍后读的保存、导入等。

**Comment by YEAHZERO** [comment]
比如这一页就不能使用阅读模式，主要是便捷保存为md文件
当然了joplin没问题

**Reply by Kenshin** [reply]
适配 Github Discussions 后就可以了，或者临时使用下图所示的方式也可以。

![image](https://user-images.githubusercontent.com/81074/166101267-aa374348-5009-4704-9aa2-f2349ecbbe03.png)

**Comment by xilin** [comment]
请问目前有办法导出标注+原文到Obsidian吗？自动化生成的html在Obsidian里识别不了。

**Reply by Kenshin** [reply]
当然没问题了，一直都可以的。

你可以看看这个教程 https://zhuanlan.zhihu.com/p/501251603

**Comment by murazi** [comment]
不使用同步助手方案

保存某些网站会失败，有办法查看日志吗？
也不知道因为什么失败了
![image](https://user-images.githubusercontent.com/4424520/183615118-11ef9476-8689-4635-a303-60301b7098d4.png)

转移到 https://github.com/Kenshin/simpread/issues/4359

**Comment by Kenshin** [comment]
嗨，各位。

简悦官方针对新用户或不喜欢配置的用户，发布了一个新功能：**Obsidan 配置库**。

细节请看下面，几分钟就能配置成完成。

***

![img](https://imgs.zhubai.love/c626678ff6b64dd1ac9dfdae6cf6d188.png)

> 针对新用户的一步到位的 Obsdian 配置库。

## 🗂 配置库

[简悦 · 配置库](https://www.yuque.com/kenshin/simpread/ds8zk0) 是简悦官方推出的一套针对新用户的极简配置方案，方便新用户用最快的方式使用简悦的各种高级服务，配置库内置了常用的双链笔记用法，如：Notion、Obsidian、Logseq、Roam Research，同时包含了简悦在阅读模式上的一些常规插件：Live Editor、题图、Safari 阅读模式等。

## 💡 特点

用户仅需要下载对应的配置库，修改里面的 UID 为自己的高级账户 UID，即可实现简悦的各种高级操作。

## 💁‍♂️ 受众

新用户或者不想使用手动配置的用户。

## 📝 如何使用

1️⃣ 下载

2️⃣ 修改为自己的 UID

3️⃣ 将必要的文件夹复制粘贴到相应位置

✅ Done！

## 📚 Obsidian 配置库

## 📝 特点

1️⃣ 加入稍后读，自动将本地快照和 Markdown 导入到 Obsidian

2️⃣ 加入标注时，自动将标注内容导入本地快照和 Obsidian

3️⃣ 改变稍后读或标注的元数据时，自动将改动的内容保存到本地快照和 Obsidian

4️⃣ 在 Obsidian 中直接内置好了简悦的标注系统，可直接在 Obsidian 实现标注

## 📖 区别

![img](https://imgs.zhubai.love/f4a51d8e8fb4401281b07f82bf0bc7d1.png)

## 📘 极速版

🔗 [Github](https://github.com/Kenshin/simpread-configs/blob/main/obsidian@little/Getting%20Started.md) | [语雀](https://www.yuque.com/kenshin/simpread/xkuecp)

不需要同步助手，无任何配置，支持 Markdown 形式的快照，自动同步标注，适合 Obsidain + 简悦的尝鲜用户。

## 📙 基础版

🔗 [Github](https://github.com/Kenshin/simpread-configs/blob/main/obsidian@localrestapi/Getting%20Started.md) | [语雀](https://www.yuque.com/kenshin/simpread/cg33gh)

需要同步助手，几乎无任何配置，支持本地快照，自动同步标注，适合 Obsidain + 简悦的轻量级用户。

## 📗 高级版

🔗 [Github](https://github.com/Kenshin/simpread-configs/blob/main/obsidian@sync/Getting%20Started.md) | [语雀](https://www.yuque.com/kenshin/simpread/wq35mh)

需要同步助手，几乎无任何配置，支持本地快照，自动同步标注，可在 Obsidian 中标注，适合 Obsidain + 简悦的重度用户。
